### ISSUE TITLE
Pyright expects overload implementations in protocols

### ISSUE THREAD

**Issue by vberlier** [microsoft/pyright#2037]
With the latest release, Pyright reports an error for the following example:

```python
class Foo(Protocol):
    @overload
    def foo(self) -> None:  # "foo" is marked as overload, but no implementation is provided
        ...

    @overload
    def foo(self, name: str) -> str:
        ...
```

**Comment by erictraut** [comment]
Thanks, I agree that an implementation shouldn't be required for a protocol class.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by vberlier** [comment]
Thank you so much!

**Comment by erictraut** [comment]
This is included in pyright 1.1.154, which I just published. It will also be included in the next release of pylance.
